test/jdk/javax/swing/JFileChooser/4525475/JFileChooserReadOnlyTest.java

```diff
@@ -0,0 +1,143 @@
+ * @run main JFileChooserReadOnlyTest
+ */
+public class JFileChooserReadOnlyTest {
+
+    private static volatile boolean result = true;
+    private static volatile boolean newFolderFound = false;
+
+    public static void main(String[] args) throws Exception {
+
+        List<String> lafs = Arrays.stream(getInstalledLookAndFeels())
+                                  .map(LookAndFeelInfo::getClassName)
+                                  .collect(Collectors.toList());
+        for (final String laf : lafs) {
+            if (!setLookAndFeel(laf)) {
+                continue;
+            }
+
+            // Test1, Read Only JFileChooser
+            final AtomicBoolean readOnly = new AtomicBoolean(true);
+            SwingUtilities.invokeAndWait(
+                    () -> createAndTestCustomFileChooser(readOnly.get()));
+            System.out.println("Its a Read Only JFileChooser " +
+                               (newFolderFound ? "but it has" :
+                                "and it doesn't have") +
+                               " a New Folder Button found" +
+                               ", So the Test1 " +
+                               (result ? "Passed" : "Failed") + " for " + laf);
+
+            // Test2, Read/Write JFileChooser
+            if (!(laf.contains("Motif") || laf.contains("Aqua"))) {
+                readOnly.set(false);
+                SwingUtilities.invokeAndWait(
+                        () -> createAndTestCustomFileChooser(readOnly.get()));
+                System.out.println("Its a not a Read Only JFileChooser " +
+                                   (newFolderFound ? "and it has" :
+                                    "but it doesn't have") +
+                                   " a New Folder Button" + ", So the Test2 " +
+                                   (result ? "Passed" : "Failed") + " for " +
+                                   laf);
+            }
+
+            if (result) {
+                System.out.println("Test Passed for " + laf);
+            } else {
+                throw new RuntimeException(
+                        "Test Failed, JFileChooser readOnly flag is not " +
+                        "working properly for " + laf);
+            }
+        }
+    }
+
+    private static void createAndTestCustomFileChooser(boolean readOnly) {
+        newFolderFound = false;
+        UIManager.put("FileChooser.readOnly", Boolean.valueOf(readOnly));
+        JFileChooser jfc = new JFileChooser();
+        checkNewFolderButton(jfc, readOnly);
+        result = readOnly ^ newFolderFound;
+    }
+
+    private static void checkNewFolderButton(Container c, boolean readOnly) {
+        int n = c.getComponentCount();
+        for (int i = 0; i < n; i++) {
+            if (newFolderFound) {
+                break;
+            }
+            Component comp = c.getComponent(i);
+            if (comp instanceof JButton) {
+                JButton b = (JButton) comp;
```

Comment:
You can use new syntax if you like:
```suggestion
            if (comp instanceof JButton b) {
```

Inside the `if` block, you can use variable `b`.

Reply:
I will keep it like this as I'm planning to backport this test to JDK 8.